zuul-core/src/main/java/com/netflix/zuul/message/http/HttpRequestMessageImpl.java

```diff
@@ -522,14 +522,17 @@ public String getOriginalProtocol()
     @Override
     public int getOriginalPort() {
         try {
-            return getOriginalPort(getHeaders(), getPort());
+            return getOriginalPort(getContext(), getHeaders(), getPort());
         } catch (URISyntaxException e) {
             throw new IllegalArgumentException(e);
         }
     }
 
     @VisibleForTesting
-    static int getOriginalPort(Headers headers, int serverPort) throws URISyntaxException {
+    static int getOriginalPort(SessionContext context, Headers headers, int serverPort) throws URISyntaxException {
+        if (context.containsKey(CommonContextKeys.REMOTE_PORT)) {
```

Comment:
Not sure if this is intended, but `REMOTE_PORT` will take precedence over the XFF header. This decision may need to be made based on whether we actually have PPv2 data or not, or even whether we prefer it.

Reply:
Thanks for catching that. I didn't realize `ATTR_REMOTE_ADDR` was available if PPv2 wasn't available. Looking now I see that it's set in `SourceAddressChannelHandler.channelActive()`. Will update.

As for precedence over XFF header, in the case of a chain of proxies either XFF or PPv2 source IP could be wrong by the time it gets to the current proxy. Not sure how precedence should be handled.